runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -191,21 +197,56 @@ private Ellipse2D drawEllipse(Graphics2D graphics, int x, int y)
 
 	private void drawSkillImage(Graphics2D graphics, XpGlobe xpGlobe, int x, int y)
 	{
-		BufferedImage skillImage = iconManager.getSkillImage(xpGlobe.getSkill());
-
-		if (skillImage == null)
+		BufferedImage toDraw;
+		if (xpGlobe.getSkillIcon() != null && xpGlobe.getGlobeSize() == config.xpOrbSize())
 		{
-			return;
+			toDraw = xpGlobe.getSkillIcon();
+		}
+		else
+		{
+			toDraw = generateScaledIcon(xpGlobe.getSkill());
+			if (toDraw == null)
+			{
+				return;
+			}
+			xpGlobe.setSkillIcon(toDraw);
+			xpGlobe.setGlobeSize(config.xpOrbSize());
 		}
 
 		graphics.drawImage(
-			skillImage,
-			x + (config.xpOrbSize() / 2) - (skillImage.getWidth() / 2),
-			y + (config.xpOrbSize() / 2) - (skillImage.getHeight() / 2),
+			toDraw,
+			x + (config.xpOrbSize() / 2) - (toDraw.getWidth() / 2),
+			y + (config.xpOrbSize() / 2) - (toDraw.getHeight() / 2),
 			null
 		);
 	}
 
+	private BufferedImage generateScaledIcon(Skill skill)
+	{
+		BufferedImage icon = iconManager.getSkillImage(skill);
+		if (icon == null)
+		{
+			return null;
+		}
+
+		final int height = (int)(config.xpOrbSize() * GlOBE_HEIGHT_RATIO);
+		final int width = (int)(config.xpOrbSize() * GLOBE_WIDTH_RATIO);
+
+		if (width == 0 || height == 0)
+		{
+			return null;
+		}
+
+		Image toolkit = icon.getScaledInstance(width, height, Image.SCALE_REPLICATE);
```

Comment:
This should just be a call to SwingUtil.resizeImage, rather than copying the code here

Reply:
Is this method supposed to be private? I have changed it to public.